toxencryptsave/data_description.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef TOXENCRYPTSAVE_DATA_DESCRIPTION_H
+#define TOXENCRYPTSAVE_DATA_DESCRIPTION_H
+
+#include "constants.hpp"
+
+namespace toxencryptsave
+{
+
+namespace offsets
+{
+    static const size_t base = 0;
+    static const size_t magic = base;
+    static const size_t salt = magic + constants::magic_len;
+    static const size_t nonce = salt + constants::salt_len;
+    static const size_t crypted_raw = nonce + constants::nonce_len;
+};
+
+template<class PointerT>
+struct Data_Description /* encryptsave data offsets + length */
```

Comment:
It is not an optimization. This is how it was before I refactored it:
```
    data += TOX_ENC_SAVE_MAGIC_LENGTH;
    data += crypto_pwhash_scryptsalsa208sha256_SALTBYTES; // salt only affects key derivation
 
    size_t decrypt_expected_length = length - TOX_PASS_ENCRYPTION_EXTRA_LENGTH;
 
    uint8_t nonce[crypto_box_NONCEBYTES];
    memcpy(nonce, data, crypto_box_NONCEBYTES);
    data += crypto_box_NONCEBYTES;
 
    /* decrypt the data */
    int decrypted_result = decrypt_data_symmetric(key->key, nonce, data, decrypt_expected_length + crypto_box_MACBYTES, out);
 
    if (decrypted_result == -1) {
        SET_ERROR_PARAMETER(error, TOX_ERR_DECRYPTION_FAILED);
        return false;
    }
```
Do you like magic offsets like that?